tools/structwrite/structwrite.go

```diff
@@ -0,0 +1,231 @@
+package structwrite
+
+import (
+	"go/ast"
+	"go/token"
+	"go/types"
+	"strings"
+
+	"github.com/golangci/plugin-module-register/register"
+	"golang.org/x/tools/go/analysis"
+)
+
+func init() {
+	register.Plugin("structwrite", New)
+}
+
+// Settings defines the configuration schema for the plugin.
+type Settings struct {
+	// Structs is a list of fully-qualified struct type names for which immutability is enforced.
+	// TODO: define in godoc instead?
+	Structs []string `json:"structs"`
+
+	// ConstructorRegex is a regex pattern (optional) to identify allowed constructor function names.
+	// TODO: unused - current just use New.*
+	ConstructorRegex string `json:"constructorRegex"`
+}
+
+// PluginStructWrite implements the LinterPlugin interface for the structwrite linter.
+type PluginStructWrite struct {
+	structs map[string]bool
```

Comment:
```suggestion
	structs map[string]struct{}
```
nit:
Are we gonna set the value to be `false`? If always be true, better to use `struct{}`, so that we know it's a set of immutable structs.

Reply:
```suggestion
	immutables map[string]bool
```

nit: IMO, Renaming to `immutables` allows the following code to be more readable:
```
	if !p.immutables[fullyQualified] {
		return
	}
```